apps/www/_blog/2023-04-10-o11y-day.mdx

```diff
@@ -0,0 +1,201 @@
+
+### Logflare Endpoint Example
+
+This example showcases a GitHub webhook that sends all event in all Supabase GitHub organizations to Logflare. This gives Logflare structured events where the payload of the Github webhook gets turned into metadata for a log event:
+
+```json
+{
+  "event_message": "supabase/supabase | JohannesBauer97 | created",
+  "id": "0d48b71d-91c5-4356-82c7-fdb299b625d0",
+  "metadata": {
+    "sender": {
+      "id": 15695124,
+      "login": "JohannesBauer97",
+      "node_id": "MDQ6VXNlcjE1Njk1MTI0",
+      "site_admin": false,
+      "type": "User",
+      "url": "https://api.github.com/users/JohannesBauer97"
+    },
+    "starred_at": "2023-03-30T20:33:55Z",
+		...
+  },
+  "timestamp": 1680208436849642
+}
+```
+
+We’re interested in the top contributors org wide, and this can be extracted by creating an Endpoint with SQL (in BigQuery dialect):
+
+```sql
+select
+  count(t.timestamp) as count,
+  s.login as gh_user
+from
+  `github.supabase.webhooks` as t
+  cross join unnest(metadata) as m
+  cross join unnest(m.sender) as s
+where
+  timestamp::date > current_date() - @day::int
+group by
+  gh_user
+order by
+  count desc
+limit
+  25
+```
+
+With this view in place, we can use Endpoints to provide an API that we can hit from our application:
+
+```bash
+curl "https://logflare.app/endpoints/query/69425db0-1cfb-48b4-84c7-2a872b6f0a61" \
+ -H 'Content-Type: application/json; charset=utf-8' \
+ -G -d "day=30"
+```
+
+Which gives us the response with the top org wide contributors for the last 30 days!
+
+```json
+{
+  "result": [
+    {
+      "count": 23404,
+      "gh_user": "vercel[bot]"
+    },
+    {
+      "count": 10005,
+      "gh_user": "joshenlim"
+    },
+    {
+      "count": 7026,
+      "gh_user": "MildTomato"
+    },
+    {
+      "count": 6405,
+      "gh_user": "fsansalvadore"
+    },
+    {
+      "count": 5195,
+      "gh_user": "saltcod"
+    },
+    {
+      "count": 3454,
+      "gh_user": "alaister"
+    },
+    {
+      "count": 2691,
+      "gh_user": "kevcodez"
+    },
+    {
+      "count": 2117,
+      "gh_user": "gregnr"
+    },
+    {
+      "count": 1769,
+      "gh_user": "Ziinc"
+    },
+    {
+      "count": 1749,
+      "gh_user": "chasers"
+    },
+    {
+      "count": 1430,
+      "gh_user": "Isaiah-Hamilton"
+    },
+    ...
+  ]
+}
+```
+
+We can then configure this Endpoint to cache results for a set interval of 10 minutes after the first API request, and proactively update those cached results every 2 minutes. This would result in 5 separate queries spread out áo cross the 10 minute cache time. Even if we were to throw tens of thousands of API requests at the endpoint, we would only sustain the cost of 5 queries to update the cached data.
+
+The initial request is fast because Logflare also performs setup (such as partitioning) on our BigQuery tables appropriately. Subsequent requests made are extremely fast as they get cached in-memory.
```

Comment:
```suggestion
The initial request is fast because Logflare also performs setup (such as partitioning) on our BigQuery tables appropriately. Subsequent requests are _extremely fast_ as they are cached in-memory.
```